--- origsrc/plasma-desktop-5.8.3/CMakeLists.txt	2016-11-01 07:19:41.000000000 -0500
+++ src/plasma-desktop-5.8.3/CMakeLists.txt	2016-11-28 03:50:26.002938200 -0600
@@ -56,7 +56,7 @@ find_package(KF5 ${KF5_MIN_VERSION} REQU
 find_package(LibKWorkspace CONFIG REQUIRED)
 find_package(LibTaskManager CONFIG REQUIRED)
 find_package(KWinDBusInterface CONFIG REQUIRED)
-find_package(ScreenSaverDBusInterface CONFIG REQUIRED)
+find_package(ScreenSaverDBusInterface CONFIG)
 find_package(KRunnerAppDBusInterface CONFIG REQUIRED)
 find_package(KSMServerDBusInterface CONFIG REQUIRED)
 find_package(KF5ItemModels CONFIG REQUIRED)
